[sdl2-image] Fails to link on arm64-osx, x64-osx and x64-linux

**Host Environment**
- OS: macOS 12.3.1
- Compiler: Apple Clang 13.1.6
- Architecture: ARM64

Testing on Linux was done on Docker using the AMD64 `ubuntu:latest` image.


**To Reproduce**
Steps to reproduce the behavior:

- Install the `sdl2-image` with either of the triplets `arm64-osx`, `x64-osx` or `x64-linux`
- Try to compile and link against a simple hello world program (fails)

<details>
<summary> CMakeLists </summary>

```cmake
cmake_minimum_required(VERSION 3.0.0)
project(Example VERSION 0.1.0)

find_package(sdl2-image CONFIG REQUIRED)

add_executable(Example main.cpp)

target_link_libraries(Example PRIVATE SDL2::SDL2_image)
```

</details>

**Failure logs**

<details>
<summary> Configure logs </summary>

```
[main] Configuring folder: blep 
[proc] Executing command: /opt/homebrew/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Release -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/clang -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/clang++ -DCMAKE_TOOLCHAIN_FILE:FILEPATH=XXX/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET:STRING=arm64-osx -DCMAKE_OSX_ARCHITECTURES:STRING=arm64 -SXXX -BXXX/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Running vcpkg install
[cmake] Detecting compiler hash for triplet arm64-osx...
[cmake] The following packages will be built and installed:
[cmake]   * libpng[core]:arm64-osx -> 1.6.37#16
[cmake]   * sdl2[core]:arm64-osx -> 2.0.20#2
[cmake]     sdl2-image[core]:arm64-osx -> 2.0.5#5
[cmake]   * vcpkg-cmake[core]:arm64-osx -> 2022-04-07
[cmake]   * vcpkg-cmake-config[core]:arm64-osx -> 2022-02-06
[cmake]   * zlib[core]:arm64-osx -> 1.2.12
[cmake] Additional packages (*) will be modified to complete this operation.
[cmake] Restored 6 packages from XXX/.cache/vcpkg/archives in 121.8 ms. Use --debug to see more details.
[cmake] Starting package 1/6: vcpkg-cmake-config:arm64-osx
[cmake] Installing package vcpkg-cmake-config[core]:arm64-osx...
[cmake] Elapsed time for package vcpkg-cmake-config:arm64-osx: 1.396 ms
[cmake] Starting package 2/6: vcpkg-cmake:arm64-osx
[cmake] Installing package vcpkg-cmake[core]:arm64-osx...
[cmake] Elapsed time for package vcpkg-cmake:arm64-osx: 2.769 ms
[cmake] Starting package 3/6: sdl2:arm64-osx
[cmake] Installing package sdl2[core]:arm64-osx...
[cmake] Elapsed time for package sdl2:arm64-osx: 25.37 ms
[cmake] Starting package 4/6: zlib:arm64-osx
[cmake] Installing package zlib[core]:arm64-osx...
[cmake] Elapsed time for package zlib:arm64-osx: 2.843 ms
[cmake] Starting package 5/6: libpng:arm64-osx
[cmake] Installing package libpng[core]:arm64-osx...
[cmake] Elapsed time for package libpng:arm64-osx: 4.743 ms
[cmake] Starting package 6/6: sdl2-image:arm64-osx
[cmake] Installing package sdl2-image[core]:arm64-osx...
[cmake] Elapsed time for package sdl2-image:arm64-osx: 3.37 ms
[cmake] 
[cmake] Total elapsed time: 2.837 s
[cmake] 
[cmake] The package sdl2-image provides CMake targets:
[cmake] 
[cmake]     find_package(sdl2-image CONFIG REQUIRED)
[cmake]     target_link_libraries(main PRIVATE SDL2::SDL2_image)
[cmake] 
[cmake] -- Running vcpkg install - done
[cmake] -- The C compiler identification is AppleClang 13.1.6.13160021
[cmake] -- The CXX compiler identification is AppleClang 13.1.6.13160021
[cmake] -- Detecting C compiler ABI info
[cmake] -- Detecting C compiler ABI info - done
[cmake] -- Check for working C compiler: /usr/bin/clang - skipped
[cmake] -- Detecting C compile features
[cmake] -- Detecting C compile features - done
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /usr/bin/clang++ - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: XXX/build
```

</details>

<details>
<summary> Build logs </summary>

```
[build] Starting build
[proc] Executing command: /opt/homebrew/bin/cmake --build XXX/build --config Release --target all --
[build] [1/2  50% :: 0.269] Building CXX object CMakeFiles/Example.dir/main.cpp.o
[build] [2/2 100% :: 0.290] Linking CXX executable Example
[build] FAILED: Example 
[build] : && /usr/bin/clang++ -O3 -DNDEBUG -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/Example.dir/main.cpp.o -o Example  vcpkg_installed/arm64-osx/lib/libSDL2_image.a  vcpkg_installed/arm64-osx/lib/libSDL2.a  -lm  -liconv  -framework  CoreVideo  -framework  Cocoa  -framework  IOKit  -framework  ForceFeedback  -framework  Carbon  -framework  CoreAudio  -framework  AudioToolbox  -framework  AVFoundation  -framework  Foundation  -Wl,-undefined,error  -Wl,-compatibility_version,1.0.0  -Wl,-current_version,19.2.0  -Wl,-weak_framework,GameController  -Wl,-weak_framework,Metal  -Wl,-weak_framework,QuartzCore  -Wl,-weak_framework,CoreHaptics  -lpng && :
[build] ld: library not found for -lpng
[build] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

</details>

**Additional context**

On maxOS, using GCC11 or LLVM13 Clang leads to the same error.

Using Unix Makefiles instead of Ninja as the build system leads to the same error.

I managed to make compilation work on macOS by editing the `sdl2-image-targets-release.cmake` (and `sdl2-image-targets-debug.cmake`). By removing in both files `png` from the `IMPORTED_LINK_INTERFACE_LIBRARIES` lists, linking works. This, however, did not work on Linux where the linking stage ended with undefined references to `libpng`.

The success on macOS is most likely a fluke as instead of linking against the statically built `libpng`, it will simply fallback to using the one in `/System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib` (as indicated when using `image list` in lldb).

The file `libpng.a` is present in the `vcpkg_installed/lib` folder:

<details>
<summary> File Tree </summary>

```
libpng.a ⇒ libpng16.a
libpng16.a
libSDL2.a
libSDL2_image.a
libSDL2main.a
libz.a
```

</details>

--- a/ports/sdl2-image/CMakeLists.txt
+++ b/ports/sdl2-image/CMakeLists.txt
@@ -68,9 +68,9 @@ if(USE_WEBP)
 endif()
 
 if(USE_PNG)
-    find_package(libpng REQUIRED)
+    find_package(PNG REQUIRED)
     add_definitions(-DLOAD_PNG)
-    target_link_libraries(SDL2_image PRIVATE png)
+    target_link_libraries(SDL2_image PRIVATE PNG::PNG)
 endif()
 
 if(USE_JPEG)

--- a/ports/sdl2-image/sdl2-image-config.cmake.in
+++ b/ports/sdl2-image/sdl2-image-config.cmake.in
@@ -1,6 +1,13 @@
 include(CMakeFindDependencyMacro)
+
 find_dependency(SDL2 CONFIG)
+
 if(@USE_WEBP@)
     find_dependency(WebP CONFIG)
 endif()
+
+if (@USE_PNG@)
+    find_dependency(PNG)
+endif()
+
 include("${CMAKE_CURRENT_LIST_DIR}/sdl2-image-targets.cmake")

--- a/ports/sdl2-image/vcpkg.json
+++ b/ports/sdl2-image/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "sdl2-image",
   "version": "2.0.5",
-  "port-version": 5,
+  "port-version": 6,
   "description": "SDL_image is an image file loading library. It loads images as SDL surfaces and textures, and supports the following formats: BMP, GIF, JPEG, LBM, PCX, PNG, PNM, TGA, TIFF, WEBP, XCF, XPM, XV",
   "homepage": "https://www.libsdl.org/projects/SDL_image",
   "dependencies": [

--- a/ports/sdl2-image/vcpkg.json
+++ b/ports/sdl2-image/vcpkg.json
@@ -4,6 +4,7 @@
   "port-version": 6,
   "description": "SDL_image is an image file loading library. It loads images as SDL surfaces and textures, and supports the following formats: BMP, GIF, JPEG, LBM, PCX, PNG, PNM, TGA, TIFF, WEBP, XCF, XPM, XV",
   "homepage": "https://www.libsdl.org/projects/SDL_image",
+  "license": "Zlib",
   "dependencies": [
     "libpng",
     "sdl2",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6378,7 +6378,7 @@
     },
     "sdl2-image": {
       "baseline": "2.0.5",
-      "port-version": 5
+      "port-version": 6
     },
     "sdl2-mixer": {
       "baseline": "2.0.4",

--- a/versions/s-/sdl2-image.json
+++ b/versions/s-/sdl2-image.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1a3da5cfed273aac0132f62b1c3acf017ff1e22e",
+      "version": "2.0.5",
+      "port-version": 6
+    },
+    {
       "git-tree": "623548e8c929f2160320bf9644e2cd5a75d4a608",
       "version": "2.0.5",
       "port-version": 5

--- a/versions/s-/sdl2-image.json
+++ b/versions/s-/sdl2-image.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "1a3da5cfed273aac0132f62b1c3acf017ff1e22e",
+      "git-tree": "e1ed993b97bc76d5a8abdf4e9e8e148903078a70",
       "version": "2.0.5",
       "port-version": 6
     },

